js/directives/daterange-picker.js

```diff
@@ -135,47 +137,44 @@
 			$scope.internal.strFriendly = $filter("luifFriendlyRange")($scope.internal);
 		};
 
-		$scope.goToPeriod = function(period){
+		$scope.goToPeriod = function(period) {
 			$scope.internal.startsOn = moment(period.startsOn).toDate();
 			$scope.internal.endsOn = moment(period.endsOn).toDate();
-			if($scope.excludeEnd){ $scope.internal.endsOn = moment(period.endsOn).add(-1,'day').toDate(); }
+			if ($scope.excludeEnd){ $scope.internal.endsOn = moment(period.endsOn).add(-1,'day').toDate(); }
 			$scope.internalUpdated();
 		};
 
 		// Popover display
 		$scope.popoverOpened = false;
-		$scope.togglePopover = function(){
+		$scope.togglePopover = function() {
 			$scope.popoverOpened = !$scope.popoverOpened;
 			if($scope.popoverOpened){
 				ctrl.pinPopover();
-			}else{
+			} else {
 				ctrl.unpinPopover();
 			}
 		};
-		$scope.doCloseAction = function(){
+
+		$scope.doCloseAction = function() {
 			$scope.togglePopover();
-			if(!!$scope.closeAction){
-				$scope.closeAction();
-			}
+			if(!!$scope.closeAction){ $scope.closeAction(); }
 		};
-		$scope.clickInside = function(e){
+
+		$scope.clickInside = function(e) {
 			e.preventDefault();
 			e.stopPropagation();
 		};
 
 		// datepickers stuff
-		$scope.dayClass = function(date, mode){
-			var className = "";
-			if(mode === "day" && moment(date).diff($scope.internal.startsOn) === 0) {
-				className = "start";
-			}
-			if(mode === "day" && moment(date).diff($scope.internal.endsOn) === 0){
-				className += "end";
-			}
-			if(mode === "day" && moment(date).isAfter($scope.internal.startsOn) && moment(date).isBefore($scope.internal.endsOn)) {
-				className += "in-between";
+		$scope.dayClass = function(date, mode) {
+			if (mode == 'day') {
+				switch(true){
+					case(moment(date).diff($scope.internal.startsOn) === 0) : return 'start';
+					case(moment(date).diff($scope.internal.endsOn) === 0) : return 'end';
+					case(moment(date).isAfter($scope.internal.startsOn) && moment(date).isBefore($scope.internal.endsOn)) : return 'in-between';
+				}				
```

Comment:
i didnt use this synthax because if `startsOn === endsOn` the date will have the class `start end`

here you can only return ` `, `start`, `end`, `in-between` and never `start end`